### ISSUE TITLE
timezone hack no longer working

### ISSUE THREAD

**Issue by lfarkas** [containers/libpod#5127]
with the latest and greatest (?!?) podman-1.7.0-2.fc31.x86_64 on fedora the old trick to set the proper timezone no longer works. until now we able to run with this eg for the official mariadb docker container:
`podman run --rm -ti --name mariadb --env-file .env -v /etc/localtime:/etc/localtime:ro --net host -v /srv/mariadb/db:/var/lib/mysql:Z -v /srv/mariadb/init:/docker-entrypoint-initdb.d:ro -v /srv/vidux/mariadb/conf:/etc/mysql/conf.d:ro mariadb:10`
but now we got this error:
`Error: mkdir ``: No such file or directory: OCI runtime command not found error`
but if we remove the localtime mount then it's start to working again ie:
`podman run --rm -ti --name mariadb --env-file .env --net host -v /srv/mariadb/db:/var/lib/mysql:Z -v /srv/mariadb/init:/docker-entrypoint-initdb.d:ro -v /srv/vidux/mariadb/conf:/etc/mysql/conf.d:ro mariadb:10`
imho it's just another regression in 1.7.0.

**Comment by rhatdan** [comment]
Is this a crun versus runc issue?  
Did this work on f31 before and now it does not?

**Comment by lfarkas** [comment]
> Is this a crun versus runc issue?

i don't know:-)

> Did this work on f31 before and now it does not?

yes it was worked before and still working on el7 but no longer on the latest podman on fedora.

**Comment by vrothberg** [comment]
@edsantiago, have you seen that?

**Comment by mheon** [comment]
What is this command supposed to do? Is it setting the timezone in the container, or on the host? What's the intended behavior, and how do we verify it is working/not working?

**Comment by edsantiago** [comment]
I wonder if this is a symlink issue? on f31 with podman-1.8.0-0.1.fc31 :
```console
# podman run -v /etc/localtime:/etc/localtime:ro alpine date
Fri Feb  7 07:29:02 MST 2020   <<< MST: expected
# podman run -v /etc/localtime:/etc/localtime:ro alpine ls -l /etc/localtime
-rw-r--r--    4 root     root          2444 Jan 19 18:38 /etc/localtime
# podman run alpine ls -l /etc/localtime
ls: /etc/localtime: No such file or directory
```
(Translation: it works just fine with alpine, which has no `/etc/localtime`)

It also works perfectly fine with `fedora`, ,which has a relative path:
```console
# podman run -v /etc/localtime:/etc/localtime:ro fedora ls -l /etc/localtime
lrwxrwxrwx. 1 root root 29 Oct 27 23:47 /etc/localtime -> ../usr/share/zoneinfo/Etc/UTC
# podman run -v /etc/localtime:/etc/localtime:ro fedora date
Fri Feb  7 07:32:54 MST 2020
```
It does not, however, work with `mariadb:10`. The container symlink does not get overriden:
```console
# podman run -v /etc/localtime:/etc/localtime:ro mariadb:10 date
Fri Feb  7 14:34:25 UTC 2020
# podman run -v /etc/localtime:/etc/localtime:ro mariadb:10 ls -l /etc/localtime
lrwxrwxrwx. 1 root root 24 Jan 16 04:37 /etc/localtime -> /usr/share/zoneinfo//UTC
```

Without delving too deeply into it, the dig difference I see is that `mariadb:10` has an absolute symlink, and I have a vague memory of lots of symlink-related work being done recently. @TomSweeneyRedHat , could this be it?

**Comment by edsantiago** [comment]
(Note: I do not see the `mkdir` error, and am ignoring all the other non-TZ-related stuff in the original post because the title specifies TZ. If there are other errors, they may need their own issues filed)

**Comment by lfarkas** [comment]
> What is this command supposed to do? Is it setting the timezone in the container, or on the host? What's the intended behavior, and how do we verify it is working/not working?

To be honest sometime I'd real doubt whether the container developer are using containers? I really wouldn't like to blame anyone but eg on fedora you can dump and restore any mysql/mariadb databases for months.
This is also a few years old hack back to docker all containers by default use UTC. To set any process run in a container you should have to run the container timezone. I you read dozens of different post about it this is one of the best solution. Of course it'd be much better if there can be a common solution for all base images (since there are at least 2 different working solution depending on the base images) but everybody use it.
But the real problem is that it was working with docker and it was working with podman up to 1.7.0. But no longer works with 1.7.0. So it's a regression.
Anyone can test it in a few minutes just using the official docker mariadb images. Probably many others I just give you one example.

**Comment by mheon** [comment]
Alright, so we're setting the container timezone by mounting in `/etc/localtime` from the host. More details would be greatly helpful - are you running as root, or without? Is this a fresh install of F31, or did you upgrade from F30? I'm sorry that this has stopped working, but fixing bugs is much easier with detailed information - that's why we have an issue template to request this.

**Comment by mheon** [comment]
I can't repro the ` Error: mkdir ``: No such file or directory: OCI runtime command not found error` on F30 or F31, root or rootless (also checked runc vs crun on F30). However, I was unable to use the exact command you provided, because I don't have any of the bind mounts you requested; do you have mariadb installed on the host, and you're mounting parts of it into the container? The error's definitely coming out of `crun` so if this is coming after a 30 to 31 upgrade, I would suspect runc to crun being the key difference.

**Comment by lfarkas** [comment]
the mkdir error no longer comes (it was with 1.7.0 as i wrote but today my system is updated to 1.8.0). anyway google gives for "container set timezone" as a first hit: https://serverfault.com/questions/683605/docker-container-time-timezone-will-not-reflect-changes.
it was a rootless podman fully updated f31 (last week).

**Comment by mheon** [comment]
Huh. Alright, we'll chalk it up to a 1.7.0 bug that got fixed in 1.8, I suppose?

I'm going to go ahead and close, but if it happens again, please re-open and we'll take another look.
